filesmanager/tasks.py

```diff
@@ -0,0 +1,65 @@
+"""
+Celery tasks for the filesmanager app.
+"""
+
+import tempfile
+import zipfile
+
+from celery import shared_task
+from celery.utils.log import get_task_logger
+from django.conf import settings
+from django.core.files.storage import default_storage
+from opaque_keys.edx.keys import AssetKey
+
+try:
+    from xmodule.contentstore.django import contentstore
+except ImportError:
+    contentstore = None
+
+logger = get_task_logger(__name__)
+
+DOWNLOADS_FOLDER = getattr(settings, "FILES_MANAGER_DOWNLOADS_FOLDER", "downloads")
+
+
+@shared_task(bind=True)
+def create_zip_file_task(self, contents):
+    """
+    Create temporary zip file with the contents passed as a parameter.
+
+    Returns the url of the zip file in the storage.
+    """
+    task_id = self.request.id
+    temporary_file = tempfile.NamedTemporaryFile(delete=False, suffix=".zip")
```

Comment:
what's `delete=False` for?

Reply:
It doesn't automatically delete the file after closed

Reply:
Updated